fdbserver/DataDistribution.actor.cpp

```diff
@@ -74,7 +74,14 @@ struct TCMachineInfo : public ReferenceCounted<TCMachineInfo> {
 	explicit TCMachineInfo(Reference<TCServerInfo> server, const LocalityEntry& entry) : localityEntry(entry) {
 		ASSERT(serversOnMachine.empty());
 		serversOnMachine.push_back(server);
-		machineID = server->lastKnownInterface.locality.zoneId().get();
+
+		LocalityData& locality = server->lastKnownInterface.locality;
+		if (locality.zoneId().present()) {
+			machineID = locality.zoneId().get();
+		} else {
+			// If locality is not set, the machine has only one server.
+			machineID = locality.processId().present() ? locality.processId().get() : StringRef(std::string(""));
+		}
```

Comment:
I think this isn't necessary now that we've confirmed that `zoneId` should always be present.

Reply:
Sure. I will change it to `ASSERT(locality.zoneId().present());`